pkg/api/controller.go

```diff
@@ -1522,10 +1522,10 @@ func handleAPIError(w http.ResponseWriter, err error) bool {
 
 	case errors.Is(err, graveler.ErrLockNotAcquired):
 		writeError(w, http.StatusInternalServerError, "branch is currently locked, try again later")
-
+	case errors.Is(err, adapter.ErrNotFound):
+		writeError(w, http.StatusGone, "Data has been removed")
```

Comment:
```suggestion
		writeError(w, http.StatusGone, "Data removed")
```
or
```suggestion
		writeError(w, http.StatusGone, "No data")
```
or even
```suggestion
		writeError(w, http.StatusGone, "Gone: removed by lifecycle")
```

"... has been" is a bit passive for an error message.